brion/plugin/compartmentReportHDF5.cpp

```diff
@@ -637,16 +757,123 @@ void CompartmentReportHDF5::_allocateDataSet()
         std::vector<size_t>{frames, _targetMapping.frameSize});
 
     HighFive::DataSetCreateProps chunking;
-    auto chunkDims = _computeChunkDims(_targetMapping.cellSizes, 5);
+    auto chunkDims =
+        _computeChunkDims(_targetMapping.cellSizes, _cellsToFramesRatio);
     if (chunkDims[0] >= frames)
-        // Only apply chunking if the dataset is large enough. Otherwise HDF5
-        // gives an error.
-        chunking.add(HighFive::Chunking(chunkDims));
+        chunkDims[0] = frames;
+    if (chunkDims[1] > _targetMapping.frameSize)
+        chunkDims[1] = _targetMapping.frameSize;
+    chunking.add(HighFive::Chunking(chunkDims));
+
+    HighFive::DataSetAccessProps caching;
+    size_t blocksPerFrame =
+        (_targetMapping.frameSize + chunkDims[1] - 1) / chunkDims[1];
+    caching.add(
+        HighFive::Caching(blocksPerFrame * _blockSize, blocksPerFrame + 1));
 
     _data.reset(new HighFive::DataSet(
-        _file->createDataSet<float>("data", dataspace, chunking)));
+        _file->createDataSet<float>("data", dataspace, chunking, caching)));
 
     detail::addStringAttribute(*_data, "dunit", _dunit);
 }
+
+void CompartmentReportHDF5::_parseWriteOptions(const URI& uri)
+{
+    for (auto i = uri.queryBegin(); i != uri.queryEnd(); ++i)
+    {
+        const auto& key = i->first;
+        const auto& value = i->second;
+        if (key == "cells_to_frames")
+        {
+            if (value == "inf")
+            {
+                _cellsToFramesRatio = std::numeric_limits<float>::infinity();
+            }
+            else
+            {
+                std::string::size_type idx;
+                const float ratio = std::stof(value, &idx);
+                if (idx != value.size())
+                {
+                    std::cerr << "Warning: invalid value for "
+                                 "cell_to_frames H5 report parameter"
+                              << std::endl;
+                }
+                else
+                {
+                    _cellsToFramesRatio = ratio;
+                }
+            }
+        }
+    }
+}
+
+void CompartmentReportHDF5::_reopenDataSet(size_t cacheSizeHint)
+{
+    // Getting the chunk dims
+    auto properties = H5Dget_create_plist(_data->getId());
```

Comment:
HighFive has no support for that?

Reply:
No, because it's a getter and we have started adding support for different properties quite recently. For the few number of lines it is I prefer not messing with HighFive.